packages/table/src/cell/editableCell.tsx

```diff
@@ -40,45 +41,81 @@ export interface IEditableCellProps extends ICellProps {
     onConfirm?: (value: string) => void;
 }
 
-export class EditableCell extends React.Component<IEditableCellProps, {}> {
-    private cellElement: HTMLElement;
+export interface IEditableCellState {
+    /**
+     * Stores the editing state of the cell
+     */
+    isEditing: boolean;
+}
+
+@PureRender
+export class EditableCell extends React.Component<IEditableCellProps, IEditableCellState> {
+    public constructor(props: IEditableCellProps, context?: any) {
+        super(props, context);
+        this.state = {
+            isEditing: false,
+        };
```

Comment:
but isn't there an issue re-using the same object for every instance? I know state objects aren't supposed to be mutated directly, but setting this in the constructor always creates a new object for the instance just to be safe. Mostly I was just following the pattern i saw elsewhere, is this not the way?